packages/calling-stateful-client/src/StreamUtils.ts

```diff
@@ -30,210 +31,214 @@ export type CreateViewResult = {
-  // stop rendering.
-  const refreshedRenderInfo = internalContext.getRemoteRenderInfoForParticipant(callId, participantKey, streamId);
-  if (!refreshedRenderInfo) {
-    // RenderInfo was removed. This should not happen unless stream was removed from the call so dispose the renderer
-    // and clean up state.
-    _logCreateRemoteStreamEvent(EventNames.REMOTE_RENDER_INFO_NOT_FOUND, streamLogInfo);
-    renderer.dispose();
-    context.setRemoteVideoStreamRendererView(callId, participantKey, streamId, undefined);
-    return;
+  if (stream) {
+    streamType = stream.mediaStreamType;
+    if (createViewType === 'remote') {
+      streamId = (stream as RemoteVideoStream).id;
+    }
   }
 
-  if (refreshedRenderInfo.status === 'Stopping') {
-    // Stop render was called on this stream after we had started rendering. We will dispose this view and do not
-    // put the view into the state.
-    _logCreateRemoteStreamEvent(EventNames.REMOTE_CREATED_STREAM_STOPPING, streamLogInfo);
-    renderer.dispose();
-    internalContext.setRemoteRenderInfo(
-      callId,
-      participantKey,
-      streamId,
-      refreshedRenderInfo.stream,
-      'NotRendered',
-      undefined
-    );
-    context.setRemoteVideoStreamRendererView(callId, participantKey, streamId, undefined);
-    return;
+  // we want to check to see if there is a participantId this will tell us whether its a local stream or a remote one.
+  let participantKey;
+  if (createViewType === 'remote' && participantId) {
+    if (typeof participantId === 'string') {
+      participantKey = participantId;
+    } else {
+      participantKey = toFlatCommunicationIdentifier(participantId);
+    }
   }
 
-  // Else the stream still exists and status is not telling us to stop rendering. Complete the render process by
-  // updating the state.
-  internalContext.setRemoteRenderInfo(
-    callId,
-    participantKey,
-    streamId,
-    refreshedRenderInfo.stream,
-    'Rendered',
-    renderer
-  );
-  context.setRemoteVideoStreamRendererView(
-    callId,
-    participantKey,
-    streamId,
-    convertFromSDKToDeclarativeVideoStreamRendererView(view)
-  );
-  _logCreateRemoteStreamEvent(EventNames.REMOTE_VIEW_RENDER_SUCCEED, streamLogInfo);
-
-  return {
-    renderer,
-    view
-  };
-}
-
-async function createViewLocalVideo(
-  context: CallContext,
-  internalContext: InternalCallContext,
-  callId: string,
-  options?: CreateViewOptions
-): Promise<CreateViewResult | undefined> {
-  // Render LocalVideoStream that is part of a Call
-  const renderInfo = internalContext.getLocalRenderInfo(callId);
-
-  const streamType = renderInfo?.stream.mediaStreamType;
+  const streamLogInfo = { callId, participantKey, streamId, streamType };
 
-  const streamLogInfo = { callId, streamType };
+  // make different logging announcement based on whether or not we are starting a local or remote
+  createViewType === 'remote'
```

Comment:
instead of every log having to check if its local or remote and having different event names, could we do:
```
const streamLogInfo = { ..., createViewType };
_logStreamEvent(EventNames.CreateView, streamLogInfo)
```